Add security configuration section to API overview

shah0108 opened the pull request:
Added a new section to the API Overview explaining how to configure nativeAllowList in neutralino.config.json. This helps developers understand how to enable specific native APIs securely.

SharonIV0x86 commented:
@shah0108 Why there is no ``nativeBlockList`` method?

SharonIV0x86 commented on docs/api/overview.md:
```diff
@@ -30,6 +30,33 @@ The client library maintains a task pool to map the server messages to the match
+    "filesystem.writeFile"
+  ]
+}
+
```
```suggestion
    ```
```
The closing backticks for json code block are missing.

shah0108 commented:
> @shah0108 Why there is no `nativeBlockList` method?

i didn’t find nativeBlockList in the neutralino.js documentation...could you clarify if it’s an existing feature?

SharonIV0x86 commented:
> > @shah0108 Why there is no `nativeBlockList` method?
> 
> i didn’t find nativeBlockList in the neutralino.js documentation...can you clarify if it is an already existing feature?

Its in documentation. 
https://neutralino.js.org/docs/configuration/neutralino.config.json#nativeblocklist-string

shah0108 commented:
yeah i checked the documentation and found nativeBlockList is a valid configuration option. i'll update the pr to include it and clarify its usage. thanks for the feedback!

shah0108 commented:
i've fixed the json formatting issue and updated the pr to include nativeBlockList along with nativeAllowList. let me know if any changes are needed, thanks

SharonIV0x86 commented on docs/api/overview.md:
```diff
@@ -30,6 +30,53 @@ The client library maintains a task pool to map the server messages to the match
+Explanation of Allowed APIs
+
+"os.*" → Grants access to all OS-related APIs.
+"filesystem.readFile" → Allows reading files.
```
```suggestion
`filesystem.readFile` → Allows reading files.
```

SharonIV0x86 commented on docs/api/overview.md:
```diff
@@ -30,6 +30,53 @@ The client library maintains a task pool to map the server messages to the match
+
+"os.*" → Grants access to all OS-related APIs.
+"filesystem.readFile" → Allows reading files.
+"filesystem.writeFile" → Allows writing files.
```
```suggestion
`filesystem.writeFile` → Allows writing files.
```

SharonIV0x86 commented on docs/api/overview.md:
````diff
@@ -30,6 +30,53 @@ The client library maintains a task pool to map the server messages to the match
+```
+Explanation of Allowed APIs
+
+"os.*" → Grants access to all OS-related APIs.
````
The explanations of these 3 allowed APIs would look better on separate lines.

SharonIV0x86 reviewed (changes requested):
Need some changes in formatting

shah0108 commented:
pls review it once again i've made the reqd changes let me know if any changes are needed, thanks!

SharonIV0x86 commented:
> pls review it once again i've made the reqd changes let me know if any changes are needed, thanks!

we can continue the discussion on the discord server.

shah0108 commented:
i gave a line gap between them so that they would look properly aligned and let me know if any changes are needed, thanks

shah0108 commented:
hey @shalithasuranga pls look through this pr and if any modifications or changes are required pls let me know, thanks!

shah0108 commented:
hey @shalithasuranga did u reviewed the pr, are there any modifications?? if there are pls let me know, thanks!?

shalithasuranga reviewed (approved):
Thanks so much @shah0108 for sending this. Please note that I've updated some things before merging this pull request :tada:

shah0108 commented:
Let's go!🚀
